app/Exceptions/Handler.php

```diff
@@ -51,75 +43,87 @@ class Handler extends ExceptionHandler
+                );
+            }
+
+            if ($e instanceof ModelNotFoundException) {
+                throw new TranslatableException(
+                    status: 404,
+                    error: $e->getMessage(),
+                    message: 'api.ERROR.MODEL_NOT_FOUND',
+                    previous: $e,
+                    shouldReport: false
                 );
             }
 
-            if ($e instanceof ModelNotFoundException || $e instanceof NotFoundHttpException) {
+            if ($e instanceof NotFoundHttpException) {
                 throw new TranslatableException(
-                    404,
-                    'Object not found (' . $e->getMessage() . ')',
-                    'api.ERROR.MODEL_NOT_FOUND',
-                    $e
+                    status: $e->getStatusCode(),
+                    error: $e->getMessage(),
+                    message: 'api.ERROR.ROUTE_NOT_FOUND',
+                    previous: $e,
+                    shouldReport: false
                 );
             }
 
             if ($e instanceof ValidationException) {
                 throw new TranslatableException(
-                    $e->status,
-                    collect($e->errors())->flatten()->implode(' '),
-                    strval(collect($e->errors())->flatten()->first()),
-                    $e,
-                    false
+                    status: $e->status,
+                    error: $e->errors(),
+                    message: $e->getMessage(),
+                    previous: $e,
+                    messageTranslatable: false,
+                    shouldReport: false,
                 );
             }
 
             if ($e instanceof HttpException && App::isDownForMaintenance()) {
                 throw new TranslatableException(
-                    $e->getStatusCode(),
-                    'Application under maintenance.',
-                    'api.ERROR.MAINTENANCE',
-                    $e
+                    status: $e->getStatusCode(),
+                    error: 'Application under maintenance.',
+                    message: 'api.ERROR.MAINTENANCE',
+                    previous: $e
                 );
             }
 
             // Generic response to other exceptions, hiding important data when debug is disabled.
             $error = get_class($e) . ' in ' . basename($e->getFile()) . ' line ' . $e->getLine() . ': ' . $e->getMessage();
 
             throw new TranslatableException(
-                method_exists($e, 'getStatusCode') && $e->getStatusCode() !== null ? $e->getStatusCode() : 500,
-                config('app.debug') && !App::runningUnitTests() ? $error : 'Server Error',
-                'api.ERROR.SOMETHING_WENT_WRONG',
-                $e
+                status: method_exists($e, 'getStatusCode') && $e->getStatusCode() !== null ? $e->getStatusCode() : 500,
+                error: config('app.debug') && !App::runningUnitTests() ? $error : $e->getMessage(),
```

Comment:
Notice the change.
- As from my experience on other projects, having lots of log items with the same 'Server Error' message is very frustrating.
- I think Exception::getMessage is intended to be a end-user allowed message (although they might not understand it at all), then privacy would not be an issue